genet/utils/graph_operations.py

```diff
@@ -282,6 +236,76 @@ def get_the_data(attributes, key):
     return data
 
 
+def build_attribute_dataframe(iterator, keys: Union[list, str], index_name: str = None):
```

Comment:
mostly copy paste of what used to be in a Network method, now reused in a bunch of places